source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown

```diff
@@ -0,0 +1,163 @@
+---
+layout: post
+title: "ESP8266 and MicroPython - Part 1"
+description: "Using MicroPython on ESP8266 based devices and Home Assistant."
+date: 2016-07-21 06:00:00 +0200
+date_formatted: "July 21, 2016"
+author: Fabian Affolter
+comments: true
+categories: How-To
+og_image: /images/blog/2016-07-micropython/social.png
+---
+
+<img src='/images/blog/2016-07-micropython/micropython.png' style='clear: right; border:none; box-shadow: none; float: right; margin-bottom: 12px;' width='200' />
+The first release of Micropython for ESP8266 was delivered a couple of weeks ago. The [documentation](http://docs.micropython.org/en/latest/esp8266/esp8266_contents.html) covers a lot of ground. This post is providing only a little summary which should get you started.
+
+If you don't participated in the KickStarter campaign only the source code was available. This meant that you needed to build the firmware on your own. As of now the [pre-built firmware](https://micropython.org/download/#esp8266) is available for the public.
+
+<!--more-->
+
+The easiest way is to use [esptool.py](https://github.com/themadinventor/esptool) for firmware handling tasks. First erase the flash:
+
+```bash
+$ sudo python esptool.py --port /dev/ttyUSB0 erase_flash
+esptool.py v1.0.2-dev
+Connecting...
+Erasing flash (this may take a while)...
+```
+
+and then load the firmware. You may adjust the file name of the firmware binary. 
+
+```bash
+$ sudo python esptool.py --port /dev/ttyUSB0 --baud 460800 write_flash --flash_size=8m 0 esp8266-2016-07-10-v1.8.2.bin
+esptool.py v1.2-dev
+Connecting...
+Running Cesanta flasher stub...
+Flash params set to 0x0020
+Writing 540672 @ 0x0... 540672 (100 %)
+Wrote 540672 bytes at 0x0 in 13.1 seconds (330.8 kbit/s)...
+Leaving...
+```
+
+Now reset the device. You should then be able to use a terminal program like `minicom` or `picocom` to connect and get the [REPL (Read Evaluate Print Loop)](http://docs.micropython.org/en/latest/esp8266/esp8266/tutorial/repl.html#getting-a-micropython-repl-prompt) prompt.
```

Comment:
On a mac: `screen /dev/tty.SLAB_USBtoUART 115200`. On Windows you can use Putty to open a serial connection. Best to link to a MicroPython page with per OS instructions ?